lib/src/workspace_store.rs

```diff
@@ -0,0 +1,214 @@
+        if !store_dir.exists() {
+            let workspaces_proto = workspace_store::Workspaces::default();
+
+            fs::create_dir(&store_dir)?;
+            fs::write(&file, workspaces_proto.encode_to_vec()).context(&file)?;
+        }
+
+        Ok(Self {
+            repo_path: repo_path.to_path_buf(),
+            store_file: file.clone(),
+            lock_file: file.with_extension("lock"),
+        })
+    }
+
+    fn lock(&self) -> Result<FileLock, WorkspaceStoreError> {
+        Ok(FileLock::lock(self.lock_file.clone()).map_err(SimpleWorkspaceStoreError::Lock)?)
+    }
+
+    fn read_store(&self) -> Result<workspace_store::Workspaces, WorkspaceStoreError> {
+        let workspace_data = fs::read(&self.store_file).context(&self.store_file)?;
+
+        let workspaces_proto = workspace_store::Workspaces::decode(&*workspace_data)
+            .map_err(SimpleWorkspaceStoreError::ProstDecode)?;
+
+        Ok(workspaces_proto)
+    }
+
+    fn write_store(
+        &self,
+        workspaces_proto: workspace_store::Workspaces,
+    ) -> Result<(), WorkspaceStoreError> {
+        let temp_file = NamedTempFile::new_in(&self.repo_path).context(&self.repo_path)?;
+
+        temp_file
+            .as_file()
+            .write_all(&workspaces_proto.encode_to_vec())
+            .context(temp_file.path())?;
+
+        persist_temp_file(temp_file, &self.store_file).context(&self.store_file)?;
+
+        Ok(())
+    }
+}
+
+impl WorkspaceStore for SimpleWorkspaceStore {
+    fn name(&self) -> &'static str {
+        "simple"
+    }
+
+    fn add(&self, workspace_name: &WorkspaceName, path: &Path) -> Result<(), WorkspaceStoreError> {
+        let _lock = self.lock()?;
+
+        let mut workspaces_proto = self.read_store()?;
+
+        workspaces_proto
+            .workspaces
+            .push(workspace_store::Workspace {
+                name: workspace_name.as_symbol().to_string(),
+                path: dunce::canonicalize(path)?.to_string_lossy().to_string(),
+            });
+
+        self.write_store(workspaces_proto)?;
+
+        Ok(())
+    }
+
+    fn forget(&self, workspace_names: &[&WorkspaceName]) -> Result<(), WorkspaceStoreError> {
+        let _lock = self.lock()?;
+
+        let mut workspaces_proto = self.read_store()?;
+
+        let workspace_names = workspace_names
+            .iter()
+            .map(|n| n.as_symbol().to_string())
+            .collect::<Vec<_>>();
+
+        workspaces_proto
+            .workspaces
+            .retain(|w| !workspace_names.contains(&w.name));
+
+        self.write_store(workspaces_proto)?;
+
+        Ok(())
+    }
+
+    fn rename(
+        &self,
+        old_name: &WorkspaceName,
+        new_name: &WorkspaceName,
+    ) -> Result<(), WorkspaceStoreError> {
+        let _lock = self.lock()?;
+
+        let mut workspaces_proto = self.read_store()?;
+
+        for workspace in &mut workspaces_proto.workspaces {
+            if workspace.name == old_name.as_symbol().to_string() {
```

Comment:
Same here: can we use `.as_str()` instead of `.as_symbol().to_string()` (and perhaps `workspace.name.as_str()` or similar)?